app/controllers/application_controller.rb

```diff
@@ -6,16 +6,63 @@ def log_in(user)
   end
 
   def log_out
+    forget(current_user)
     session.delete(:user_id)
     @current_user=nil
   end
 
+  def forget(user)
+    user.forget
+    cookies.delete(:user_id)
+    cookies.delete(:remember_token)
+  end 
+
+
+  helper_method :current_user
+  def current_user
+    if (user_id = session[:user_id])
+      @current_user ||= User.find_by(id: user_id)
+    elsif (user_id = cookies.signed[:user_id])
```

Comment:
Curious why you're using both `session` and `cookies` for storing user session info?